### PR TITLE
[jsk_pcl_ros] Add sample_color_histogram_matcher.launch

### PR DESCRIPTION
**Pull request by taichiH**
Add color histogram matcher sample launch file becauce of https://github.com/jsk-ros-pkg/jsk_recognition/pull/2428

Modification
1. add `sample_color_histogram_matcher.launch` and `play_rosbag_convenience_store.xml`
2. add `color_histogram_publisher.py` to publish color_histogram topic constantly.
3. modified `install_sample_data.py` to install sample rosbag file.
4. add rviz config file for visualization.

### CODE REVIEW

File: jsk_pcl_ros/sample/sample_color_histogram_matcher.launch [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  <arg name="gui" default="true" />
+
+  <include file="$(find jsk_pcl_ros)/sample/include/play_rosbag_convenience_store.xml" />
+
+  <node name="color_histogram_publisher"
+        pkg="jsk_pcl_ros" type="color_histogram_publisher.py"
+        output="screen">
+    <rosparam>
+      frame_id: base_link
+      histogram: [
```
Comment (YutoUchimi) [inline jsk_pcl_ros/sample/sample_color_histogram_matcher.launch]: Please also run `jsk_pcl_ros/color_histogram_visualizer.py` and show this reference histogram in rviz.

File: jsk_pcl_ros/sample/sample_color_histogram_matcher.launch [hunk 2]
```diff
@@ -0,0 +1,44 @@
+<launch>
```
Comment (YutoUchimi) [inline jsk_pcl_ros/sample/sample_color_histogram_matcher.launch]: Can you also update `Sample` in `color_histogram_matcher.md`?

Comment (YutoUchimi) [inline jsk_pcl_ros/sample/sample_color_histogram_matcher.launch]: Can you add test?
Only testing if output topics are published is ok.

**Comment by k-okada** [comment]
nice sample program. please explain how to get `histogram` parameter https://github.com/jsk-ros-pkg/jsk_recognition/pull/2429/files#diff-c083d243c1549987c0b6f490a61323b2R11

**Comment by taichiH** [comment]
I was also wondering how to get a color histogram.

In my case, 
1. Get a specific region point cloud from cluster point indices.
2. Input extracted point cloud to color_histogram_mather
3. I can get color_histogram as an output_reference topic from a color_histogram_mather node.

But this is weird because this way needs a color_histogram_mather node for color_histogram_mather node.

So far as I know, there is not a useful converter from something(e.g. bounding box ..) to a color histogram in jsk_recognition.
As a result, I got a histogram by a strange method described above.

**Comment by k-okada** [comment]
that's very important information

1) create new issue "need a smart way to obtain color_histgram"
2) add a comment to existing launch file "please refer to Issue #xx on how
to get color histogram value"

I think you can create small simple launch file that takes xyz:={0, 0, 0}
size:={1,1,1} as input and publish histogram. If that uses
color_histogram_matcher node inside, the user will feel it is very useful
tools

--
◉ Kei Okada



2019年5月29日(水) 19:05 Taichi Higashide <notifications@github.com>:

**Comment by knorth55** [comment]
When I register new color histogram, I run color histogram node, subscribe a topic and copy & paste it.

File: jsk_pcl_ros/sample/include/play_rosbag_convenience_store.xml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<launch>
+  <param name="/use_sim_time" value="false" />
+
+  <arg name="filename" default="$(find jsk_pcl_ros)/sample/data/convenience_store.bag" />
```
[the remark this answers is not in the document]
Comment (taichiH) [reply jsk_pcl_ros/sample/include/play_rosbag_convenience_store.xml]: Sorry, that is my fault, fixed rosbag file that has consistent TF tree.

File: jsk_pcl_ros/sample/sample_color_histogram_matcher.launch (hunk 2 above)
Comment (taichiH) [reply jsk_pcl_ros/sample/sample_color_histogram_matcher.launch]: I updated document.

Comment (taichiH) [reply jsk_pcl_ros/sample/sample_color_histogram_matcher.launch]: I added test for a color_histogram_matcher node, I confirmed correctly pass a rostest on my environment.

File: jsk_pcl_ros/sample/sample_color_histogram_matcher.launch (hunk 1 above)
Comment (taichiH) [reply jsk_pcl_ros/sample/sample_color_histogram_matcher.launch]: I think sample launch file should keep as simple as possible.
So, I'd like to reduce the ros node even just a little bit.
But If there are some of the coding rules, I'll follow them.

File: jsk_pcl_ros/sample/rviz/color_histogram_matcher.rviz [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+      Focal Point:
+        X: -0.0262356717
+        Y: 0.044948779
+        Z: 0.995900989
+      Focal Shape Fixed Size: false
+      Focal Shape Size: 0.0500000007
+      Invert Z Axis: true
+      Name: Current View
+      Near Clip Distance: 0.00999999978
+      Pitch: 1.56979632
```
[the remark this answers is not in the document]
Comment (taichiH) [reply jsk_pcl_ros/sample/rviz/color_histogram_matcher.rviz]: I fixed it.

**Comment by taichiH** [comment]
We can get a color histogram by combining attention clipper with color histogram nodes.
I create a simple launch file that input is point cloud and bounding box, the output is color histogram array.
I create a simple launch file to extract color histogram by bounding box.\
I think this is also `ColorHistogram` node sample.

```
header: 
  seq: 1740
  stamp: 
    secs: 1561114591
    nsecs:  34519426
  frame_id: "openni_camera_rgb_optical_frame"
histograms: 
  - 
    header: 
      seq: 1759
      stamp: 
        secs: 1561114591
        nsecs:  34519426
      frame_id: "openni_camera_rgb_optical_frame"
    histogram: [0.10601483285427094, 0.09887393563985825, 0.002471848390996456, 0.0013732491061091423, 0.00357044767588377, 0.04669047147035599, 0.3004668951034546, 0.021697336807847023, 0.00878879427909851, 0.03405657783150673, 0.3194177448749542, 0.05657786503434181]

```
color_histogram_extractor

**Pull request merged**